plotly/Make_a_FigureFactory.md

```diff
@@ -0,0 +1,113 @@
+from plotly.figure_factory._candlestick import create_candlestick
+...
+```
+
+If you want to make, for example, a chart called `foo`, then you must create a python file `_foo.py` and then add the following line to the end of `__init__.py`:
+
+```
+from plotly.figure_factory._foo import create_foo
+```
+
+2. Imports
+
+In `_foo.py` write
+
+```
+from __future__ import absolute_import
+```
+
+at line 1. You can add other imports later if you will need them.
+
+3. The main function
+
+It's now time to write the main function `create_foo` that will be called directly by the user. It has the form:
+
+```
+def create_foo(data, height=450, width=600, ...):
+    """
+    Returns figure for a foo plot.
+
+    :param (list) data: description of what 'data' is.
+    :param (int) height: description of what 'height' is.
+    :param (int) width: description of what 'width' is.
+    # ...
+    
+    Example 1:
+    '''
+    
+    '''
+    
+    Example 2:
+    '''
+    
+    '''
+    """
+    # code
+    # ...
+    # return fig
+```
+
+You _must_ include what is known as a documentation string or doc string in your function, which is just a block string taht contains useful information about what the function does, the arguments of the function and their descriptions, and examples of this function in use. The doc string is displayed when the help method is run by a user: `help(create_foo)` or `create_foo?` in Jupyter.
+
+The parameters are listed in the doc string with the format `:param (param_type) param_name: description.` Afterwards, you must include Examples which demonstrate the different capabilities and features of the function. For more information on proper doc string syntax see [PEP-257 page](https://www.python.org/dev/peps/pep-0257/).
+
+After the doc string, you may write the main code of your function, which should result in returning the `fig`. Users will use your function in the following way:
+
+```
+# create figure
+fig = create_foo(...)
+
+# plot figure
+py.iplot(fig, filename='my_figure')
+```
+
+The figure `fig` must be a Plotly Figure, meaning it must have the form `fig = graph_objs.Figure(data=data, layout=layout)`.
+
+4. Useful Tips
+
+It is often not a good idea to put all your code into your `create_foo()` function. It is best practice to not repeat yourself and this requires taking repeated blocks of code and puting them into a seperate function. Usually it is best to make all other functions besides `create_foo()` secret so a user cannot access them. This is done by placing a `_` before the name of the function, so `_aux_func()` for example.
+
+
+## Push to GitHub
+
+When you are finally finished your first draft of your figure factory, it is time to push it to the cloud and to get feedback from the Plotly team and other voluntary GitHub users. After you have added and commited all of your changes on the local branch, push the changes to a new remote branch on Git:
```

Comment:
not 100% sure what:
>it is time to push it to the cloud 
means,
shouldn't we explain how to create a pr. They want to push their code to their remote github branch.

Reply:
When you have finished the first draft of your figure factory, it is time to create a pull request for the Plotly team to review.